test/extended/cluster/utils.go

```diff
@@ -41,12 +48,58 @@ func ParsePods(jsonFile string) (configStruct kapiv1.Pod) {
 	return
 }
 
+// Wait for pods to go into Running or "not Running" state
+func SyncPods(c kclientset.Interface, ns string, selectors map[string]string, timeout time.Duration, negate bool) (err error) {
```

Comment:
This has already been implemented a bunch of times in the e2e framework ([example](https://github.com/kubernetes/perf-tests/blob/master/clusterloader/framework/pod_utils.go#L47-L64)). Do we want another version here?

Reply:
So how do you implement waiting for a pod to go into anything but "Running" state using the code you pointed out?

Reply:
Usually we would expect a specific state.

Reply:
I do not see why.  Expecting anything but "not Running" instead of "Complete/Error/..." is a valid reason for synchronization.